src/resolve/resolved-dns-transaction.c

```diff
@@ -523,7 +544,7 @@ static int dns_transaction_maybe_restart(DnsTransaction *t) {
 static void on_transaction_stream_error(DnsTransaction *t, int error) {
         assert(t);
 
-        dns_transaction_close_connection(t);
+        dns_transaction_close_connection(t, true);
```

Comment:
(not overly familiar with resolved) if this happens on error, shouldn't it be false? Or is the error not related to the UDP connection?

Reply:
If you look into dns_transaction_close_connection() you'll see that it only passes the UDP socket to the graveyard if it's open. it will never pass the TCP socket (which is kept in another variable).

Or to say this differently: the boolean argument to dns_transaction_close_connection() has no effect if no UDP socket is open anyway.

Reply:
Ah, missed that this was about the TCP one